python/ray/_private/runtime_env/plugin.py

```diff
@@ -158,73 +172,86 @@ def load_plugins(self, plugin_configs: List[Dict]):
-                    "it should be an integer between "
-                    f"{RAY_RUNTIME_ENV_PLUGIN_MIN_PRIORITY} "
-                    f"and {RAY_RUNTIME_ENV_PLUGIN_MAX_PRIORITY}."
-                )
-
-            self.plugins[plugin_class.name] = RuntimeEnvPluginManager.Context(
-                plugin_class(), priority
+            self.validate_priority(priority)
+
+            class_instance = plugin_class()
+            self.plugins[plugin_class.name] = PluginSetupContext(
+                plugin_class.name,
+                class_instance,
+                priority,
+                self.create_uri_cache_for_plugin(class_instance),
             )
 
-    def sorted_plugin_setup_contexts(
-        self, inputs: List[Tuple[str, Any]]
-    ) -> List[PluginSetupContext]:
-        used_plugins = []
-        for name, config in inputs:
-            if name not in self.plugins:
-                default_logger.error(
-                    f"runtime_env field {name} is not recognized by "
-                    "Ray and will be ignored.  In the future, unrecognized "
-                    "fields in the runtime_env will raise an exception."
-                )
-                continue
-            used_plugins.append(
-                (
-                    name,
-                    config,
-                    self.plugins[name].class_instance,
-                    self.plugins[name].priority,
-                )
-            )
-        sort_used_plugins = sorted(used_plugins, key=lambda x: x[3], reverse=False)
-        return [
-            PluginSetupContext(name, config, class_instance)
-            for name, config, class_instance, _ in sort_used_plugins
-        ]
+    def add_plugin(self, plugin: RuntimeEnvPlugin) -> None:
+        """Add a plugin to the manager and create a URI cache for it.
 
+        Args:
+            plugin: The class instance of the plugin.
+        """
+        plugin_class = type(plugin)
+        self.validate_plugin_class(plugin_class)
+        self.validate_priority(plugin_class.priority)
+        self.plugins[plugin_class.name] = PluginSetupContext(
+            plugin_class.name,
+            plugin,
+            plugin_class.priority,
+            self.create_uri_cache_for_plugin(plugin),
+        )
+
+    def create_uri_cache_for_plugin(self, plugin: RuntimeEnvPlugin) -> URICache:
+        """Create a URI cache for a plugin.
 
-@DeveloperAPI
-class PluginCacheManager:
-    """Manages a plugin and a cache for its local resources."""
+        Args:
+            plugin_name: The name of the plugin.
 
-    def __init__(self, plugin: RuntimeEnvPlugin, uri_cache: URICache):
-        self._plugin = plugin
-        self._uri_cache = uri_cache
+        Returns:
+            The created URI cache for the plugin.
+        """
+        # Set the max size for the cache.  Defaults to 10 GB.
+        cache_size_env_var = f"RAY_RUNTIME_ENV_{plugin.name}_CACHE_SIZE_GB".upper()
```

Comment:
In long term, we should move this env var to the json config(maybe a yaml config file in future) of runtime env plugin.